### PR TITLE
Fixes #8437: Recent changes graph can overflow the page width

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/8437

### CODE REVIEW

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala [hunk 1]
```diff
@@ -227,10 +227,14 @@ class RuleEditForm(
             $$( "#editRuleZone" ).tabs('select', ${tab});"""
           )) &
           JsRaw(s"""
-
             $$("#editRuleZonePortlet").removeClass("nodisplay");
             ${Replace("details", new RuleCompliance(rule, rootRuleCategory).display).toJsCmd};
             scrollToElement("${idToScroll}", ".rudder_col");
+            $$("#editRuleZone").bind( "show", function(event, ui) {
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala]: 'var recentChart' is not defined before

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala [hunk 2]
```diff
@@ -227,10 +227,14 @@ class RuleEditForm(
...
+              if(ui.panel.id== 'ruleComplianceTab') {
+                recentChart.flush();
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala]: Before flushing, you should check that recentChart is not undefined, unless you will have a js error when calling flush on 'undefined'

**Comment by RaphaelGauthier** [comment]
PR updated

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala (hunk 1 above)
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala]: Ok, i didn't know that you could 'declare' global variables by doing directly "xxx =" without var before so ok !

File: rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala [hunk 3]
```diff
@@ -227,10 +227,14 @@ class RuleEditForm(
             $$( "#editRuleZone" ).tabs('select', ${tab});"""
           )) &
           JsRaw(s"""
-
             $$("#editRuleZonePortlet").removeClass("nodisplay");
             ${Replace("details", new RuleCompliance(rule, rootRuleCategory).display).toJsCmd};
             scrollToElement("${idToScroll}", ".rudder_col");
+            $$("#editRuleZone").bind( "show", function(event, ui) {
+              if((ui.panel.id== 'ruleComplianceTab')&&(typeof recentChart !== 'undefined')) {
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/RuleEditForm.scala]: you should not use typeof and only do:

```
recentChart !== undefined
```

**Comment by RaphaelGauthier** [comment]
PR updated

**Pull request merged**